entrypoint.sh

```diff
@@ -2,262 +2,227 @@
-            exit 0 
-        else 
-            new=$(semver -i "${default_semvar_bump}" "$tag")
-            part=$default_semvar_bump 
-        fi 
-        ;;
-esac
-
-if $pre_release
-then
-    # get current commit hash for tag
-    pre_tag_commit=$(git rev-list -n 1 "$pre_tag")
-    # skip if there are no new commits for pre_release
-    if [ "$pre_tag_commit" == "$commit" ]
-    then
-        echo "No new commits since previous pre_tag. Skipping..."
-        setOutput "new_tag" "$pre_tag"
-        setOutput "tag" "$pre_tag"
-        exit 0
-    fi
-    # already a pre-release available, bump it
-    if [[ "$pre_tag" =~ $new ]] && [[ "$pre_tag" =~ $suffix ]]
-    then
-        if $with_v
-        then
-            new=v$(semver -i prerelease "${pre_tag}" --preid "${suffix}")
-        else
-            new=$(semver -i prerelease "${pre_tag}" --preid "${suffix}")
-        fi
-        echo -e "Bumping ${suffix} pre-tag ${pre_tag}. New pre-tag ${new}"
-    else
-        if $with_v
-        then
-            new="v$new-$suffix.0"
-        else
-            new="$new-$suffix.0"
-        fi
-        echo -e "Setting ${suffix} pre-tag ${pre_tag} - With pre-tag ${new}"
-    fi
-    part="pre-$part"
+# Determine semver bump
+determine_bump() {
+	local bump_type
+
+	case "$log" in
+	*$major_string_token*)
+		bump_type="major"
+		;;
+	*$minor_string_token*)
+		bump_type="minor"
+		;;
+	*$patch_string_token*)
+		bump_type="patch"
+		;;
+	*$none_string_token*)
+		return 1
+		;;
+	*)
+		if [ "$default_semver_bump" == "none" ]; then
+			return 1
+		else
+			bump_type="${default_semver_bump}"
+		fi
+		;;
+	esac
+
+	semver -i "${bump_type}" "$tag"
+}
+
+new_tag=$(determine_bump)
+if [ $? -eq 1 ]; then
+	echo "Default bump was set to none. Skipping..."
+	setOutput "new_tag" "$tag"
+	setOutput "tag" "$tag"
+	exit 0
+fi
+
+if $pre_release; then
+	if [[ $pre_tag =~ $new_tag ]] && [[ $pre_tag =~ $prerelease_suffix ]]; then
+		new_tag=$([[ $with_v ]] && echo "v$(semver -i prerelease "${pre_tag}" --preid "${prerelease_suffix}")" || echo "$(semver -i prerelease "${pre_tag}" --preid "${prerelease_suffix}")")
+		echo -e "Bumping ${prerelease_suffix} pre-tag ${pre_tag}. New pre-tag ${new_tag}"
+	else
+		new_tag=$([[ $with_v ]] && echo "v$new_tag-$prerelease_suffix.0" || echo "$new_tag-$prerelease_suffix.0")
+		echo -e "Setting ${prerelease_suffix} pre-tag ${pre_tag} - With pre-tag ${new_tag}"
+	fi
 else
-    if $with_v
-    then
-        new="v$new"
-    fi
-    echo -e "Bumping tag ${tag} - New tag ${new}"
+	new_tag=$([[ $with_v ]] && echo "v$new_tag" || echo "$new_tag")
+	echo -e "Bumping tag ${tag} - New tag ${new_tag}"
 fi
 
-# as defined in readme if CUSTOM_TAG is used any semver calculations are irrelevant.
-if [ -n "$custom_tag" ]
-then
-    new="$custom_tag"
+if [ -n "$custom_tag" ]; then
+	new_tag="$custom_tag"
 fi
 
-# set outputs
-setOutput "new_tag" "$new"
-setOutput "part" "$part"
-setOutput "tag" "$new" # this needs to go in v2 is breaking change
```

Comment:
I would like this comment to remain specially since broke the flows for a LOT lots of people